monkey/monkey_island/cc/repository/file_agent_configuration_repository.py

```diff
@@ -25,10 +26,10 @@ def get_configuration(self) -> AgentConfiguration:
                 configuration_json = f.read().decode()
 
             return self._schema.loads(configuration_json)
-        # TODO: Handle FileRetrievalError vs FileNotFoundError
-        #       https://github.com/guardicore/monkey/blob/e8001d8cf76340e42bf17ff62523bd2d85fc4841/monkey/monkey_island/cc/repository/file_storage/local_storage_file_repository.py#L47-L50
-        except RetrievalError:
+        except repository.FileNotFoundError:
             return self._schema.loads(DEFAULT_AGENT_CONFIGURATION)
```

Comment:
1. The default configuration must be coupled to something, otherwise nothing can fetch or serve it. Removing this coupling from the repository doesn't reduce overall coupling within the system, it just moves it from the repository to somewhere else. We could reduce this to some extent by passing the default configuration into the repository's constructor.
2. In practice, this doesn't mask any errors. If we add a default at startup, then the interface behaves the same way it's currently defined, that is, when you call `get()`, you get back the default (because we added it at startup). We add temporal coupling and complexity to the user, without actually making anything more explicit.
3. "Check if there's no config and insert the default" on startup doesn't solve temporal coupling, it is temporal coupling. Everything that uses this would be forced to handle the case where no default was added at startup.

It's a trade-off. Like any trade-off, we'll need to evaluate it each time we use it.